server/src/test/java/org/elasticsearch/cluster/routing/allocation/allocator/DesiredBalanceComputerTests.java

```diff
@@ -1215,33 +1215,37 @@ private void checkIterationLogging(int iterations, long eachIterationDuration, M
         var clusterSettings = createBuiltInClusterSettings(
             Settings.builder().put(DesiredBalanceComputer.MAX_BALANCE_COMPUTATION_TIME_DURING_INDEX_CREATION_SETTING.getKey(), "2m").build()
         );
-        var desiredBalanceComputer = new DesiredBalanceComputer(clusterSettings, mockThreadPool, new ShardsAllocator() {
-            @Override
-            public void allocate(RoutingAllocation allocation) {
-                final var unassignedIterator = allocation.routingNodes().unassigned().iterator();
-                while (unassignedIterator.hasNext()) {
-                    final var shardRouting = unassignedIterator.next();
-                    if (shardRouting.primary()) {
-                        unassignedIterator.initialize("node-0", null, 0L, allocation.changes());
-                    } else {
-                        unassignedIterator.removeAndIgnore(UnassignedInfo.AllocationStatus.NO_ATTEMPT, allocation.changes());
+        var desiredBalanceComputer = new DesiredBalanceComputer(
+            clusterSettings,
+            mockThreadPool::relativeTimeInMillis,
```

Comment:
It looks like this mockThreadPool could be simplified as well
```suggestion
            () -> currentTime.addAndGet(eachIterationDuration),
```
this way it might not be created at all above.

Reply:
oh, right. didn't see that. thanks.